app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -242,6 +246,7 @@ class WallpapersUseCases(
             }
             val currentWallpaper = possibleWallpapers.find { it.name == currentWallpaperName }
                 ?: fileManager.lookupExpiredWallpaper(currentWallpaperName)
+                ?: migrationHelper.migrateLegacyWallpaper(currentWallpaperName)
```

Comment:
Rather than include this here, I think it would make sense to run the migration separately after the current wallpaper is dispatched on line 240. We could also wrap the call in a conditional based on a property in `Settings` so that we only do it once. I believe I saw that included in a previous version of this patch. Something like:

```
Wallpaper.getCurrentWallpaperFromSettings(settings)?.let {
    store.dispatch(AppAction.WallpaperAction.UpdateCurrentWallpaper(it))
}
val currentWallpaperName = withContext(Dispatchers.IO) { settings.currentWallpaperName }
if (settings.requiresWallpaperMigration) {
    migrationHelper.migrateLegacyWallpaper(currentWallpaperName)
}
```

Once the wallpaper is migrated, it should get picked up automatically as part of `possibleWallpapers` or `lookupExpiredWallpaper`. 

My thinking is that this change will make it easier to remove the migration code in the future. Also, it should make it more clear that the migration code is only running once, rather than relying on the conditional in the migration code that looks like `if (!legacyLandscapeFile.exists() || !legacyPortraitFile.exists())`. This will also skip the IO of looking for the legacy files.